AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreConsecutiveUpdatesTests.swift

```diff
@@ -0,0 +1,432 @@
+
+        let hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent else {
+                XCTFail("Can't cast payload as mutation event")
+                return
+            }
+
+            guard let post = try? mutationEvent.decodeModel() as? Post, post.id == newPost.id else {
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                XCTAssertEqual(post, newPost)
+                saveSyncReceived.fulfill()
+                return
+            }
+
+            if mutationEvent.mutationType == GraphQLMutationType.delete.rawValue {
+                XCTAssertEqual(post, newPost)
+                // this needs to be commented out once the bug is fixed
+                // deleteSyncReceived.fulfill()
+                return
+            }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // save the post
+        let saveAndImmediatelyDelete = expectation(description: "Post is saved and then immediately deleted")
+        Amplify.DataStore.save(newPost) { result in
+            switch result {
+            case .success:
+                Amplify.DataStore.delete(newPost) { result in
+                    switch result {
+                    case .success:
+                        saveAndImmediatelyDelete.fulfill()
+                    case .failure(let error):
+                        XCTFail("Error: \(error)")
+                    }
+                }
+            case .failure(let error):
+                XCTFail("Error: \(error)")
+            }
+        }
+        wait(for: [saveAndImmediatelyDelete], timeout: networkTimeout)
+
+        // query the deleted post immediately
+        let queryResult = queryPost(byId: newPost.id)
+        XCTAssertNil(queryResult)
+
+        wait(for: [saveSyncReceived], timeout: networkTimeout)
+        // this needs to be commented out once the bug is fixed
+        // wait(for: [deleteSyncReceived], timeout: networkTimeout)
+
+        // query the deleted post in eventual consistent state
+        let queryResultAfterSync = queryPost(byId: newPost.id)
+
+        // this should be removed once the bug is fixed
+        XCTAssertNotNil(queryResultAfterSync)
+        XCTAssertEqual(queryResultAfterSync, newPost)
+
+        // this is the actual behavior which is currently failing
+        // XCTAssertNil(post)
+
+        // API query
+        let queryRequest =
+            GraphQLRequest<MutationSyncResult?>.query(modelName: newPost.modelName, byId: newPost.id)
+        let apiQuerySuccess = expectation(description: "API query is successful")
+        Amplify.API.query(request: queryRequest) { result in
```

Comment:
API should be checking the response that the deleted field is true, if the test currently fails then put the check and comment it out and add comment on why it is commented out